Consul.Test/ConnectTest.cs

```diff
@@ -76,5 +80,75 @@ public async Task Connect_CASetConfig()
             Assert.Equal("bar", updatedConfig.State["foo"]);
             Assert.Equal("", updatedConfig.Config["PrivateKey"]);
         }
+
+
+        [Fact]
+        public async Task Connect_ListIntentions()
+        {
+
+            var firstEntry = new ServiceIntentionsEntry
+            {
+                Kind = "service-intentions",
+                Name = "Autobots-Assembler",
+                Sources = new List<SourceIntention>
+                {
+                    new SourceIntention
+                    {
+                        Name = "fortunate",
+                        Action = "allow"
+                    },
+                    new SourceIntention
+                    {
+                        Name = "Prad",
+                        Action = "allow"
+                    },
+                    new SourceIntention
+                    {
+                        Name = "Medhi",
+                        Action = "allow"
+                    }
+                }
+            };
+
+            var secondEntry = new ServiceIntentionsEntry
+            {
+                Kind = "service-intentions",
+                Name = "Test",
+                Sources = new List<SourceIntention>
+                {
+                    new SourceIntention
+                    {
+                        Name = "Optimus-Prime",
+                        Action = "allow"
+                    },
+                    new SourceIntention
+                    {
+                        Name = "Megatron",
+                        Action = "deny"
+                    }
+                }
+
+            };
+
+
+            var resultOne = await _client.Configuration.ApplyConfig(firstEntry);
+            Assert.Equal(HttpStatusCode.OK, resultOne.StatusCode);
+
+            var resultTwo = await _client.Configuration.ApplyConfig(secondEntry);
+            Assert.Equal(HttpStatusCode.OK, resultTwo.StatusCode);
+
+            var intentionsQuery = await _client.Connect.ListIntentions<ServiceIntentionsEntryResponse>();
+            Assert.Equal(HttpStatusCode.OK, intentionsQuery.StatusCode);
+
+            var intentions = intentionsQuery.Response;
+            Assert.NotNull(intentions);
+
+
+            Assert.Contains(intentions, i => i.SourceName == "fortunate");
+            Assert.Contains(intentions, i => i.DestinationName == firstEntry.Name);
```

Comment:
Please make the test more comprehensive by testing all the members of an intention.
If the exact expected value is unknown, we can at least test if all fields are non-null or non-zero.